http/http-api/src/main/java/com/okta/commons/http/config/HttpClientConfiguration.java

```diff
@@ -37,6 +39,85 @@
     private int retryMaxElapsed = 0;
     private int retryMaxAttempts = 0;
 
+    private Integer maxConnectionPerRoute;
+    private Integer maxConnectionTotal;
+    private Integer connectionValidationInactivity;
+    private Integer connectionTimeToLive;
+
+    private static final Logger log = LoggerFactory.getLogger(HttpClientConfiguration.class);
+
+    private static final String MAX_CONNECTIONS_PER_ROUTE_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.maxPerRoute";
+    private static final int MAX_CONNECTIONS_PER_ROUTE_DEFAULT_VALUE = Integer.MAX_VALUE/2;
+    private static final String MAX_CONNECTIONS_PER_ROUTE_WARNING_MESSAGE = "Bad max connection per route value";
+
+    private static final String MAX_CONNECTIONS_TOTAL_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.maxTotal";
+    private static final int MAX_CONNECTIONS_TOTAL_DEFAULT_VALUE = Integer.MAX_VALUE;
+    private static final String MAX_CONNECTIONS_TOTAL_WARNING_MESSAGE = "Bad max connection total value";
+
+    private static final String CONNECTION_VALIDATION_INACTIVITY_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.validateAfterInactivity";
+    private static final int CONNECTION_VALIDATION_INACTIVITY_DEFAULT_VALUE = 2000; // 2sec
+    private static final String CONNECTION_VALIDATION_INACTIVITY_WARNING_MESSAGE = "Invalid max connection inactivity validation value";
+
+    private static final String CONNECTION_TIME_TO_LIVE_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.timeToLive";
+    private static final int CONNECTION_TIME_TO_LIVE_DEFAULT_VALUE = 5 * 1000 * 60; // 5 minutes
+    private static final String CONNECTION_TIME_TO_LIVE_WARNING_MESSAGE = "Invalid connection time to live value";
+
+    public String getMaxConnectionsTotalPropertyKey() {
+        return MAX_CONNECTIONS_TOTAL_PROPERTY_KEY;
+    }
+
+    public String getMaxConnectionsPerRoutePropertyKey() {
+        return MAX_CONNECTIONS_PER_ROUTE_PROPERTY_KEY;
+    }
+
+    public Integer getMaxConnectionPerRouteDefault() {
+        return MAX_CONNECTIONS_PER_ROUTE_DEFAULT_VALUE;
+    }
+
+    public Integer getMaxConnectionTotalDefault() {
+        return MAX_CONNECTIONS_TOTAL_DEFAULT_VALUE;
+    }
```

Comment:
These methods should not be public, they are just static values.  The goal is to remove the static configuration (i.e. the System.property calls from the static block.)

Reply:
Updated